include/deal.II/grid/manifold.h

```diff
@@ -397,6 +400,33 @@ class Manifold : public Subscriptor
                  const std::vector<double>           &weights) const;
 
   /**
+   * Compute a new set of points around the given points
+   * @p surrounding_points. @p weights is a table with as many columns as
+   * @p surrounding_points.size(). The number of rows in @p weights determines
+   * how many new points will be computed and appended to the last input
+   * argument @p new_points. After exit of this function, the size of
+   * @p new_points equals the size at entry plus the number of rows in
+   * @p weights.
+   *
+   * In its default implementation, this function simply calls
+   * get_new_point() on each row of @weights and appends those points to the
+   * output vector @p new_points. However, this function is more efficient if
+   * multiple new points need to be generated like in MappingQGeneric and the
+   * manifold does expensive transformations between a chart space and the
+   * physical space, such as ChartManifold. If efficiency is not important,
+   * you may get away by implementing only the get_new_point() function.
+   *
+   * The implementation does not allow for @p surrounding_points and
+   * @p new_points to point to the same vector, so make sure pass different
+   * objects into the function.
```

Comment:
The problem is due to the way I implement `Manifold::get_new_points`: Inside that function, we repeatedly call `get_new_points(surrounding_points, weight)` and of course this does not work if `surrounding_points` keeps increasing in size with every point we append to that array. We could of course put a copy function to the generic `add_new_points` in case the pointers are equal. Do you prefer that?